src/app/AttributePathParams.h

```diff
@@ -81,10 +83,49 @@ struct AttributePathParams
         return true;
     }
 
-    ClusterId mClusterId     = kInvalidClusterId;   // uint32
-    AttributeId mAttributeId = kInvalidAttributeId; // uint32
-    EndpointId mEndpointId   = kInvalidEndpointId;  // uint16
-    ListIndex mListIndex     = kInvalidListIndex;   // uint16
+    bool IsAttributePathIntersect(const DataVersionFilter & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+
+        return true;
+    }
+
+    bool IsAttributePathIntersect(const ConcreteClusterPath & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+
+        return true;
+    }
+
+    bool IsWildcardRequest(const AttributePathParams & aOther) const
+    {
+        if (HasWildcardEndpointId())
+        {
+            return true;
+        }
+        if (HasWildcardClusterId())
+        {
+            return true;
+        }
+        if (mEndpointId == aOther.mEndpointId && mClusterId == aOther.mClusterId && HasWildcardAttributeId())
+        {
+            return true;
+        }
+        return false;
+    }
+
+    bool operator<(const AttributePathParams & aOther) const
```

Comment:
Where are we using this `operator<`?

Reply:
we are creating std::sort for mRequestPathSet, which needs < to be implemented

Reply:
The use of `std::set` is fine to whittle out duplicate entries, but the actual operator< should not be defined here, since it's very specific to a particular use-case with the AttributeCache.

If it were a generalized comparison, you would have incorporated AttributeId and ListIndex but you didn't.

Reply:
Agree. Would move it out

Reply:
done, thanks